openmc/tallies.py

```diff
@@ -304,24 +317,30 @@ def nonzero(val):
                 return 1 if not val else val
 
             # Reshape the results arrays
-            new_shape = (nonzero(self.num_filter_bins),
-                         nonzero(self.num_nuclides),
-                         nonzero(self.num_score_bins))
-
-            sum = np.reshape(sum, new_shape)
-            sum_sq = np.reshape(sum_sq, new_shape)
+            sum = np.reshape(sum, self.shape)
+            sum_sq = np.reshape(sum_sq, self.shape)
```

Comment:
These are changes pulled in from @samuelshaner's PR, so perhaps he can
remove nonzero there. I'm waiting to fully review that PR a second time
till he gives the go ahead. It does a good job generalizing tally
arithmetic, and you may want to review it as well since the changes are
discussuon-worthy. In any case, most of the diff here will be removed once
we merge in that PR.
On Dec 14, 2015 8:52 AM, "Paul Romano" <notifications@github.com> wrote:

> In openmc/tallies.py
> <https://github.com/mit-crpg/openmc/pull/528#discussion_r47499049>:
>
> > @@ -304,24 +317,30 @@ def nonzero(val):
> >                  return 1 if not val else val
> >
> >              # Reshape the results arrays
> > -            new_shape = (nonzero(self.num_filter_bins),
> > -                         nonzero(self.num_nuclides),
> > -                         nonzero(self.num_score_bins))
> > -
> > -            sum = np.reshape(sum, new_shape)
> > -            sum_sq = np.reshape(sum_sq, new_shape)
> > +            sum = np.reshape(sum, self.shape)
> > +            sum_sq = np.reshape(sum_sq, self.shape)
>
> No need for nonzero anymore? Looks like it's unused now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/528/files#r47499049>.
>